### ISSUE TITLE
Possible bug on 'can.Model.store' attribute implementation

### ISSUE THREAD

**Issue by meconcomputer** [canjs/canjs#2112]
When migrating an old project in JVMC 3.2, I noticed that the following chain of actions, give erroneous results:

```javascript
can.fixture('GET /a', function() {
return {
   a: 1,
   c: {
		id: 'ID1',
		shouldNotBeInModelB: true,
		anotherFieldInA: 2
   }
};
})

can.fixture('GET /b', function() {
return {
   b: 1,
   c: {
		id: 'ID1',
		fieldInB: 2
   }
};
})

can.Model('C', {
model: function(obj){
   var retval = can.Model.model.apply(C, arguments);
   console.log(retval, retval.shouldNotBeInModelB);
   return retval;
}
}, {})

can.Model('A',{
	findOne: 'GET /a'
},{
	define: {
		c: {
			type: C.model
		}
	}
})

can.Model('B', {
	findOne: 'GET /b'
}, {
   define: {
		c: {
			type: C.model
		}
   }
})
```

The results are the following: 
A.findOne() returns
```javascript
{
   anotherFieldInA: 2
   id: "ID1"
   shouldNotBeInModelB: true
}
```
But B.findOne() returns
```javascript
{ 
   anotherFieldInA: 2
   fieldInB: 2
   id: "ID1"
   shouldNotBeInModelB: true
}
```
erroneous result merging result from A.findOne() and the correct result from B.findOne().

It might seem an odd example, but it is perfectly "legal" according to documentation.

Looking through source code, it seems that the cause is the current implementation of `store` attribute (see `can/model/model.js:128`).

What is the real cause of this behaviour?

**Comment by rjgotten** [comment]
> What is the real cause of this behaviour?

This is expected behavior.

You're fetching a `can.Model` with the same identity ("ID1") twice. The second fetch will take the already existing instance and update it with the newly received properties and values.

If you don't want this behavior, make C a `can.Map` instead of a `can.Model`.

**Comment by meconcomputer** [comment]
Yes, I understand that. But it is a common thing to have two different items (obtained from different endpoints), but with the same id, isn't it?

**Comment by meconcomputer** [comment]
For example two tables, `cats`, `dogs`, having references to a common `genre` table. By chance, when getting a `cat` and a `dog` having the same genre `male` id, but backend processing adds different attributes to `male` genre.
This is the situation the above example is used.

**Comment by rjgotten** [comment]
> But it is a common thing to have two different items (obtained from different endpoints), but with the same id

They're not different. In both cases the type of the member `c` is `C`.

**Comment by pYr0x** [comment]
@meconcomputer, rjgotten is right.

you have to create a new instance of C
```
can.Model('A',{
    findOne: 'GET /a'
},{
    define: {
        c: {
            type: function(newValue){
              return new C(newValue);
            }
        }
    }
});
```
example:
http://jsbin.com/recewopiwi/edit?html,js,console


for type converting see also this good post at: http://forums.donejs.com/t/managing-complex-string-fields-with-can-map/64

**Comment by meconcomputer** [comment]
@pYr0x great resource donejs forum. My bad, I didn't searched enough.
@rjgotten thank you for your replies.
Just to be clearer on the subject. I thought that `store` had an issue using 2 endpoints, returning an object (with the same id), which was then passed to a the `model` method of `can.Model`. Each endpoint processed data differently so even if the 2 endpoints used the same model, I was getting erroneous results.
@pYr0x I didn't want to use define `{ Type: C }`, because I had a custom `C.model`, so it was necessary to pass through this method
In the end I used @rjgotten observations, and for one of the enpoints, I used another model for `c` attribute, extended from `C` model.
